src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.TargetingPackResolution.targets

```diff
@@ -0,0 +1,150 @@
+                                TargetFrameworkVersion="$(_TargetFrameworkVersionWithoutV)"
+                                TargetingPackRoot="$(NetCoreTargetingPackRoot)"
+                                AppHostRuntimeIdentifier="$(AppHostRuntimeIdentifier)"
+                                RuntimeGraphPath="$(BundledRuntimeIdentifierGraphFile)"
+                                SelfContained="$(SelfContained)"
+                                RuntimeIdentifier="$(RuntimeIdentifier)"
+                                DotNetAppHostExecutableNameWithoutExtension="$(_DotNetAppHostExecutableNameWithoutExtension)">
+
+      <Output TaskParameter="PackagesToDownload" ItemName="PackageReferenceToAdd" />
+      <Output TaskParameter="PackagesToReference" ItemName="PackageReference" />
+      <Output TaskParameter="RuntimeFrameworks" ItemName="RuntimeFramework" />
+      <Output TaskParameter="TargetingPacks" ItemName="TargetingPack" />
+      <Output TaskParameter="AppHost" ItemName="AppHostPack" />
+
+      <Output TaskParameter="UnresolvedFrameworkReferences" ItemName="_UnresolvedFrameworkReference" />
+
+    </ResolveFrameworkReferences>
+
+    <ItemGroup>
+      <PackageReference Include="@(PackageReferenceToAdd)"
+                        IsImplicitlyDefined="true"
+                        PrivateAssets="all"
+                        ExcludeAssets="all" />
+    </ItemGroup>
+
+  </Target>
+
+  <UsingTask TaskName="ReportUnknownFrameworkReferences" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+
+  <!-- Report unresolved FrameworkReferences in a separate target so that the error is only reported on Build, not Restore. -->
+  <Target Name="ReportUnresolvedFrameworkReferences" DependsOnTargets="ResolveFrameworkReferences"
+          BeforeTargets="_CheckForInvalidConfigurationAndPlatform"
+          Condition="'@(_UnresolvedFrameworkReference)' != ''">
+
+    <!-- Use a custom task in order to report multiple errors instead of stopping on the first one (which would happen if we batched) -->
+    <ReportUnknownFrameworkReferences UnresolvedFrameworkReferences="@(_UnresolvedFrameworkReference)"/>
+
+  </Target>
+
+
+  <UsingTask TaskName="GetPackageDirectory" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+  
+  <Target Name="ResolveTargetingPacks" DependsOnTargets="ResolvePackageAssets">
+
+    <GetPackageDirectory
+      Items="@(TargetingPack)"
+      ProjectPath="$(MSBuildProjectFullPath)"
+      PackageFolders="@(AssetFilePackageFolder)">
+
+      <Output TaskParameter="Output" ItemName="ResolvedTargetingPack" />
+      
+    </GetPackageDirectory>    
+    
+    <ItemGroup>
+      <ResolvedTargetingPack>
+        <Path Condition="'%(ResolvedTargetingPack.Path)' == ''">%(PackageDirectory)</Path>
+      </ResolvedTargetingPack>
+      <Reference Include="%(ResolvedTargetingPack.Path)\ref\netcoreapp3.0\*.dll">
+        <!-- TODO: Work out what metadata we should use here to display these references grouped under the targeting pack
```

Comment:
Please file bug(s) for TODOs.

Reply:
Filed https://github.com/dotnet/sdk/issues/2802 and https://github.com/dotnet/project-system/issues/4444